control/mgcp/src/main/java/org/restcomm/media/core/control/mgcp/endpoint/MediaGroupImpl.java

```diff
@@ -77,11 +87,15 @@ private Player initializePlayer(PlayerProvider players) {
         return player;
     }
 
-    private DtmfDetector initializeDetector(DtmfDetectorProvider detectors) {
-        // TODO try getting rid of DetectorImpl cast
-        DetectorImpl detector = (DetectorImpl) detectors.provide();
-        this.audioComponent.addOutput(detector.getAudioOutput());
-        this.oobComponent.addOutput(detector.getOOBOutput());
+    private DtmfSinkFacade initializeDetector(DtmfDetectorProvider detectors) {
+        AudioOutput output = new AudioOutput(detectors.getScheduler(), ComponentType.DTMF_DETECTOR.getType());
```

Comment:
This doesn't seem right. The details of building the Facade don't belong here.

I'd recommend switching the DtmfDetectorProvider with a DtmfFacadeProvider and place this logic inside that class. It will also help managing dependency injection.